### PR TITLE
[Bug](regression) fix bug of null in topn opt

### PR DESCRIPTION
**Pull request by HappenLee**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by HappenLee** [comment]
run buildall

File: be/src/olap/accept_null_predicate.h [hunk 1]
```diff
@@ -116,29 +118,32 @@ class AcceptNullPredicate : public ColumnPredicate {
...
+        } else {
+            _nested->evaluate_vec(column, size, flags);
         }
     }
 
     void evaluate_and_vec(const vectorized::IColumn& column, uint16_t size,
                           bool* flags) const override {
         if (column.has_null()) {
             // copy original flags
-            auto original_flags_buf = std::make_unique<bool[]>(size);
```
Comment (BiteTheDDDDt) [inline be/src/olap/accept_null_predicate.h:134]: better do not change it

**Review by BiteTheDDDDt — APPROVED** [review]

**Review by eldenmoon — APPROVED** [review]
LGTM

File: be/src/olap/accept_null_predicate.h [hunk 2]
```diff
@@ -168,44 +173,35 @@ class AcceptNullPredicate : public ColumnPredicate {
...
-            // init to 0 / false
-            memset(selected_flags, 0, (max_idx + 1) * sizeof(bool));
-            for (uint16_t i = 0; i < size; ++i) {
-                uint16_t row_idx = sel[i];
-                if (column.is_null_at(row_idx)) {
-                    // set selected flag true for NULL value
-                    selected_flags[row_idx] = true;
-                }
-            }
+            uint16_t max_idx = sel[size - 1];
```
Comment (xiaokang) [inline be/src/olap/accept_null_predicate.h:178]: It's more reasonable to use *std::max_element(sel, sel + size)

**Review by xiaokang — APPROVED** [review]
LGTM

**Pull request merged**